### ISSUE TITLE
Reproduction of NYU Depth results gets different numbers?

### ISSUE THREAD

**Issue by tyleryzhu** [EPFL-VILAB/MultiMAE#26]
Hello! I'm trying to reproduce the NYU depth results but getting much higher delta_1's than shown in the paper (but also potentially elsewhere). As you can see below attached, I've run it with three different backbones (DEiT from timm, MAE, and MultiMAE pre-trained weights) and gotten a best delta_1 of around 89. I followed the NYU dataset preparation steps, but seems like something may have gone wrong. Any tips would be appreciated. Perhaps the masking is incorrect?

<img width="1474" alt="Screen Shot 2023-02-16 at 11 47 23 PM" src="https://user-images.githubusercontent.com/22010655/219582725-f8de06c1-f596-4be8-b14c-bb46375da308.png">

**Comment by AndrewAtanov** [comment]
Hello @tyleryzhu,

Could you share some more details about your fine-tuning settings? Do you use [run_finetuning_depth.py](https://github.com/EPFL-VILAB/MultiMAE/blob/main/run_finetuning_depth.py) for NYU depth fine-tuning from this repo? If not:
- Do you follow the details outlined in Appendix B.3 of the paper and hyper-parameters from the Table 9? See also [the config](https://github.com/EPFL-VILAB/MultiMAE/blob/main/cfgs/finetune/depth/ft_nyu_2000e_multimae-b.yaml).
- Do you fine-tune with the [reversed Huber loss](https://github.com/EPFL-VILAB/MultiMAE/blob/66910f5b5ba236f5e731883db85fe4f24ee01106/run_finetuning_depth.py#L69)?
- Does you implementation of the delta_1 metric matches the one [here](https://github.com/EPFL-VILAB/MultiMAE/blob/66910f5b5ba236f5e731883db85fe4f24ee01106/run_finetuning_depth.py#L113)?

Andrei

**Comment by tyleryzhu** [comment]
Hi @AndrewAtanov, 

Thank you for the reply! I'm using the default ```run_finetuning_depth.py``` as you mentioned with the default settings given in the config in the repo for mae/mmae. For those, I am using the berhu loss, as well as the same delta_1 eval code. The only significant difference I can see is that I am changing my batch size to fit on my GPUs. Do you have any logs you could share so I can match the perf? I am most curious that I am able to get a much higher delta_1 that described in the paper.

**Comment by AndrewAtanov** [comment]
Yes, this is indeed interesting that you got a higher score. I have attached the config and logs from the corresponding run so that you can double-check them. Do other losses/lr/ look similar? Which checkpoint do you use for fine-tuning?

[nyu_depth_logs.zip](https://github.com/EPFL-VILAB/MultiMAE/files/10831107/nyu_depth_logs.zip)

**Comment by tyleryzhu** [comment]
Thanks for the logs! The config seems to be empty, could you update with that if you have it? I'm using the MultiMAE checkpoint provided. Interestingly my val losses are a lot higher than yours, so maybe there's a mismatch there. I will look into it more.

**Comment by AndrewAtanov** [comment]
oh, indeed, here is the updated archive.
[nyu_depth_logs.zip](https://github.com/EPFL-VILAB/MultiMAE/files/10840082/nyu_depth_logs.zip)

**Comment by Huiimin5** [comment]
> Hello! I'm trying to reproduce the NYU depth results but getting much higher delta_1's than shown in the paper (but also potentially elsewhere). As you can see below attached, I've run it with three different backbones (DEiT from timm, MAE, and MultiMAE pre-trained weights) and gotten a best delta_1 of around 89. I followed the NYU dataset preparation steps, but seems like something may have gone wrong. Any tips would be appreciated. Perhaps the masking is incorrect?
> 
> <img alt="Screen Shot 2023-02-16 at 11 47 23 PM" width="1474" src="https://user-images.githubusercontent.com/22010655/219582725-f8de06c1-f596-4be8-b14c-bb46375da308.png">

Hi, did you figure out why? I also got a better finetuning results for depth estimation than those reported in the paper. I checked the config and I did not see a difference in hyperparameters that influences the fineturning process.

**Comment by AndrewAtanov** [comment]
Hi @tyleryzhu @Huiimin5,

We reevaluated the checkpoints used for reporting the NYU Depth results and, indeed, found that the performance is higher for all the cases. For MultiMAE, we got a similar result of `88.1` for delta1. It is unclear what the exact cause of the initial mistake was as we are using the same checkpoints as before, but we will issue an updated paper shortly. 

Note that you should also increase `max_val_images` (e.g., `--max_val_images 1000`), as was pointed out in #28. This not the cause of the difference as 100 val images give similar results now.

Thanks for pointing this out! I'll close the issue, but feel free to reopen it if there is more.
